uat/component_publish.feature

```diff
@@ -0,0 +1,62 @@
+Feature: gdk component publish works
+
+  @version(min='1.0.0')
+  @change_cwd
+  Scenario: publish template zip
+    Given we have cli installed
+    And we make directory helloworld
+    And we run init command with -t,HelloWorld,-l,python
+    And command was successful
+    And we verify component files
+    And change component name to com.example.PythonHelloWorld
+    And change artifact uri for all platform from HelloWorld to helloworld
+    And we run build command
+    And command was successful
+    And we verify component zip build files
+    And we verify build artifact helloworld.zip
+    When we run publish command
+    And command was successful
+
+  @version(min='1.1.0')
+  @change_cwd
+  Scenario: publish template zip with directory name
+    Given we have cli installed
+    And we run init command with -t,HelloWorld,-l,python,-n,HelloWorld
+    And command was successful
+    And we change directory HelloWorld
+    And we verify component files
+    And change component name to com.example.PythonHelloWorld
+    And we run build command
+    And command was successful
+    And we verify component zip build files
+    And we verify build artifact HelloWorld.zip
+    When we run publish command
+    And command was successful
+
+  @version(min='1.1.0')
+  @change_cwd
+  Scenario: publish template zip with bucket arg
+    Given we have cli installed
+    And we run init command with -t,HelloWorld,-l,python,-n,HelloWorld
+    And command was successful
+    And we change directory HelloWorld
+    And we verify component files
+    And change component name to com.example.PythonHelloWorld
+    And we get s3 bucket name
+    When we run publish command with -b,<last_s3_bucket>
+    And command was successful
+    And we verify component zip build files
+    And we verify build artifact HelloWorld.zip
+
+  @version(gt='1.1.0')
+  @change_cwd
+  Scenario: publish template maven multi project with mixed uris
+    Given we have cli installed
+    And we download zip file maven-mixed-uris-publish-test from s3
+    And we unpack that zip file into current directory
+    And we change directory maven-mixed-uris-publish-test
```

Comment:
Nit: Though this is exactly what we do in the tests, it'd be nice to extract everything out as 
`And we have gdk project maven-mixed-uris-publish-test` which internally would call all these three. As through UATs, we want to show what exactly a customer does.

Reply:
We can abstract these away if these feel too granular, but gdk project setup can done from different mechanisms, so that needs to be clear. 

How about `we setup gdk project named maven-mixed-uris-publish-test from s3`?

Reply:
Yup, nmakes sense.

Reply:
done